streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -229,29 +230,29 @@ public void testCloseIsIdempotent() throws Exception {
             closeCount, MockMetricsReporter.CLOSE_COUNT.get());
     }
 
-    @Test(expected = IllegalStateException.class)
+    @Test
     public void testCannotStartOnceClosed() throws Exception {
         streams.start();
         streams.close();
         try {
             streams.start();
-        } catch (final IllegalStateException e) {
-            Assert.assertEquals("Cannot start again.", e.getMessage());
-            throw e;
+            fail("Should have thrown IllegalStateException.");
```

Comment:
Hi Matthias,
I think overall using @Rule annotations is a cleaner approach compared to the proposed solution - unless it's not applicable here for some reason I don't see.
More precisely I'm thinking about using ExpectedException like this:
```
 public class SimpleExpectedExceptionTest {
     @Rule
     public ExpectedException thrown = ExpectedException.none();

     @Test
     public void throwsNothing() {
         // no exception expected, none thrown: passes.
     }

     @Test
     public void throwsExceptionWithSpecificType() {
         thrown.expect(NullPointerException.class);
         throw new NullPointerException();
     }
 }
```
What do you think?

Reply:
I guess that would work too. ATM, we use the "fail-pattern" throughout the whole code base. \cc @dguy WDYT? Should we start to transit to a "rule-pattern"?